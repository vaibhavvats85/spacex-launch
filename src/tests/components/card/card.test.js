import { shallow } from 'enzyme';
import Card from '../../../components/card/card';
import React from 'react';

const mockPrograms = {
    flight_number: 1,
    mission_name: 'SpaceX first',
    mission_id: ['9AD2ER', '1GT234'],
    launch_year: '2020',
    launch_success: 'true',
    rocket: {
        first_stage: {
            cores: [
                {
                    land_success: null
                }
            ]
        }
    },
    links: {
        mission_patch_small: 'mock-image'
    }
};

let wrapper;

beforeEach(() => {
    wrapper = shallow(<Card data={mockPrograms} />);
});

test('should render card correctly', () => {
    expect(wrapper).toMatchSnapshot();
});