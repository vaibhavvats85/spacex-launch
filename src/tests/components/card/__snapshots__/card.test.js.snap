// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render card correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-img"
  >
    <img
      alt="rocket"
      src="mock-image"
    />
  </div>
  <span
    className="title"
  >
    SpaceX first
     #
    1
  </span>
  <div
    className="card-body"
  >
    <div>
      <strong>
        Mission Ids
        : 
      </strong>
      <ul
        className="value"
      >
        <li
          key="9AD2ER"
        >
          9AD2ER
        </li>
        <li
          key="1GT234"
        >
          1GT234
        </li>
      </ul>
    </div>
    <div>
      <strong>
        Launch Year
        : 
      </strong>
      <span
        className="value"
      >
        2020
      </span>
    </div>
    <div>
      <strong>
        Successful Launch
        : 
      </strong>
      <span
        className="value"
      >
        true
      </span>
    </div>
    <div>
      <strong>
        Successful Landing
        : 
      </strong>
      <span
        className="value"
      >
        false
      </span>
    </div>
  </div>
</div>
`;
